R/01-createNewDBSource.R

```diff
@@ -0,0 +1,168 @@
+#' Create New DB Source
+#'
+#' Creates a script for a new data source from a database connection and sets .Renviron variables.
+#' Only "mySql" databases are supported.
+#'
+#' @param dbName (character) name of the new database source, e.g. "xyDBname"
+#' @param tableName name of the table containing the data
+#' @param descriptionCreator (character) command that creates the description, e.g. pasting data
+#'  columns "var1" and "var2": "paste(isoData$var1, isoData$var2)"
+#' @param datingType (character) dating type for the database
+#' @param coordType (character) coord type for the database
+#' @param scriptFolder (character) place to store the scripts.
+#' @param rootFolder (character) root folder of the package, usually containing .Renviron,
+#' DESCRIPTION, ...
+createNewDBSource <- function(dbName,
+                              tableName,
+                              datingType,
+                              coordType,
+                              descriptionCreator = NULL,
+                              scriptFolder = "R",
+                              rootFolder = "") {
+  # check for duplicated db names
+  if (toupper(dbName) %in% toupper(dbnames()))
+    stop(paste0("dbName = ", dbName, " already exists in (",
+                paste0(dbnames(), collapse = ", "),
+                "). Please provide case-insensitive unique names."))
+
+  dbScript <- pasteScriptBegin(dbName = dbName)
+
+  dbScript <- c(dbScript,
+                addDataLoadForDB(dbName = dbName))
+
+  dbScript <- c(
+    dbScript,
+    addDescription(descriptionCreator = descriptionCreator),
+    pasteScriptEnd()
+  )
+
+  dbScript <- c(dbScript,
+                addDBSettings(dbName = dbName,
+                              tableName = tableName))
+
+  writeLines(dbScript, con = file.path(scriptFolder, paste0("02-", dbName, ".R")))
+
+  setupRenviron(dbName = dbName, scriptFolder = file.path(rootFolder))
+
+  updateDatabaseList(
+    dbName = dbName,
+    datingType = datingType,
+    coordType = coordType,
+    scriptFolder = file.path(scriptFolder)
+  )
+}
+
+
+#' Add Data Load For database source
+#'
+#' add lines to the script for data load from database
+#'
+#' @inheritParams createNewDBSource
+addDataLoadForDB <- function(dbName) {
+  c("# load data",
+    paste0("  isoData <- get", dbName, "()"),
+    "")
+}
+
+
+#' Paste DB settings
+#'
+#' @inheritParams createNewDBSource
+addDBSettings <- function(dbName, tableName) {
```

Comment:
I find these sort of tasks much easier to read and understand when we use a templating system. For dbtools and SQL Queries we use templates. E.g. see https://cran.r-project.org/web/packages/templates/vignettes/Template_Programming_in_R.html

I am the author of dbtools and templates and obviously biased, please make a change only if you see it as an improvement.

In case you want to use templates, you may also want to put the template into its own file. Like `./inst/templates/db-settings.R`. You can then reliably refer to that file with `system.file`.

Reply:
Now, I am using templates. Thanks for that remark @wahani ! I added 3 new templates to `./inst/templates`